### PR TITLE
fix: make the `Series.apply` outcome `assign`able to the original dataframe in partial ordering mode

### PR DESCRIPTION
**Pull request by shobsi**
BEGIN_COMMIT_OVERRIDE
fix: make the `Series.apply` outcome `assign`able to the original dataframe in partial ordering mode (#874)

chore: apply remote_function on the original series without reprojecting (#874)
END_COMMIT_OVERRIDE

This change tests application of remote function without reprojecting the original series.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

**Comment by shobsi** [comment]
was only for testing the need for force reproject

**Review by TrevorBergeron — APPROVED** [review]

File: bigframes/functions/_remote_function_session.py [hunk 1]
```diff
@@ -499,6 +505,20 @@ def try_delattr(attr):
...
             )
 
+            # TODO(shobs): Find a better way to support udfs with param named "name".
+            # This causes an issue in the ibis compilation.
+            func.__signature__ = inspect.signature(func).replace(  # type: ignore
+                parameters=[
+                    inspect.Parameter(
+                        f"bigframes_{param.name}"
+                        if param.name == "name"
+                        else param.name,
```
[the remark this answers is not in the document]
Comment (shobsi) [reply bigframes/functions/_remote_function_session.py]: Done, thanks!

**Comment by tswast** [comment]
@shobsi Does this make `remote_function` more usable from "partial ordering mode"? If so, I think it should go in the changelog as a feat/fix.

**Comment by shobsi** [comment]
> @shobsi Does this make `remote_function` more usable from "partial ordering mode"? If so, I think it should go in the changelog as a feat/fix.

Makes sense, added COMMIT OVERRIDE, should be picked up by the next release-please PR.

**Pull request merged**